File: Avaliacao.java
public class Avaliacao {
    private Estudante aluno;
    private float nota;

    public Avaliacao(Estudante aluno, float nota) {
        this.aluno = aluno;
        this.nota = nota;
    }

    Estudante obterAluno() {
        return aluno;
    }

    float obterNota() {
        return nota;
    }

    @Override
    public String toString() {
        return "Aluno: " + aluno.obterMatricula() + " - Nota: " + nota;
    }
}

File: cpp_version/Modelo/Avaliacao.cpp
#include "Avaliacao.h"

Avaliacao :: Avaliacao (Estudante* aluno, double nota){
    this->aluno = new Estudante(aluno->obterMatricula(), aluno->obterNome());
    this->nota = nota;
}

Avaliacao :: ~Avaliacao(){
    delete aluno;
}

Estudante* Avaliacao :: obterAluno(){
    return aluno;
}

double Avaliacao :: obterNota(){
    return nota;
}

ostream& operator<< (ostream& os, Avaliacao &avaliacao){
    os << "Aluno: " << avaliacao.aluno->obterMatricula() << " - Nota: " << avaliacao.nota << endl;
    return os;
}